src/game/SpellEffects.cpp

```diff
@@ -3700,15 +3714,19 @@ void Spell::DoSummon(SpellEffectIndex eff_idx)
 
     if (m_caster->GetTypeId() == TYPEID_PLAYER)
     {
-        // this enables popup window (pet dismiss, cancel)
         spawnCreature->SetUInt32Value(UNIT_FIELD_FLAGS, UNIT_FLAG_PVP_ATTACKABLE);
 
-        spawnCreature->GetCharmInfo()->SetPetNumber(pet_number, true);
+        CharmInfo* charmInfo = spawnCreature->GetCharmInfo();
+        if (m_caster->getClass() == CLASS_PRIEST)
```

Comment:
Yes, the more in sync the better; I'll go for your check. All I know is that these types of pets weren't introduced before tbc so they do not exist at all in vanilla.